### PR TITLE
[_]: feature/file-endpoints

### PR DESCRIPTION
**Pull request by sraver**
Moved remaining endpoints on the storage controller mostly related to files management.

### CODE REVIEW

File: src/app/routes/storage.ts [hunk 1]
```diff
@@ -285,6 +289,268 @@ export class StorageController {
...
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Error updating metadata from file ${fileId} : ${err}`);
+        res.status(500).json(err.message);
+      });
+  }
+
+  public async deleteFileBridge(req: Request, res: Response): Promise<void> {
+    if (req.params.bucketid === 'null') { // Weird checking...
```
Comment (sg-gs) [inline src/app/routes/storage.ts:368]: For God's sake

File: src/app/routes/storage.ts [hunk 2]
```diff
@@ -285,6 +289,268 @@ export class StorageController {
...
+        res.status(404).end();
+      });
+  }
+
+  public async renameFileInNetwork(req: Request, res: Response): Promise<void> {
+    const { behalfUser: user } = req as SharedRequest;
+    const { bucketId, fileId, relativePath } = req.body;
+    const mnemonic = req.headers['internxt-mnemonic'];
+
+    if (Validator.isEmpty(fileId)) { // Not sure the datatype of it. number or string
```
Comment (sg-gs) [inline src/app/routes/storage.ts:513]: Is a string, because this fileId is the network file id

File: tests/controllers/storage.test.ts [hunk 3]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        body: {
+          metadata: {},
+        },
+        headers: {}
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.updateFile(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1567]: `expect(true).to.be.false`

File: tests/controllers/storage.test.ts [hunk 4]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+          bucketId: '',
+          relativePath: 'sss',
+        },
+        headers: {}
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.updateFile(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1593]: same here

File: tests/controllers/storage.test.ts [hunk 5]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+          bucketId: 'ss',
+          relativePath: '',
+        },
+        headers: {}
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.updateFile(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1619]: and here

File: tests/controllers/storage.test.ts [hunk 6]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+          bucketId: 'ss',
+          relativePath: 'sss',
+        },
+        headers: {}
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.updateFile(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1645]: and here

File: tests/controllers/storage.test.ts [hunk 7]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        status: () => {
+          return {
+            json: jsonSpy
+          };
+        }
+      });
+
+      try {
+        // Act
+        await controller.deleteFileDatabase(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1915]: and here

File: tests/controllers/storage.test.ts [hunk 8]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:1944]: and here

File: tests/controllers/storage.test.ts [hunk 9]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        status: () => {
+          return {
+            json: jsonSpy
+          };
+        }
+      });
+
+      try {
+        // Act
+        await controller.getRecentFiles(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2065]: and here

File: tests/controllers/storage.test.ts [hunk 10]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        params: {
+          folderId: '',
+          lockId: '',
+        },
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.acquireFolderLock(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2229]: and here

File: tests/controllers/storage.test.ts [hunk 11]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        params: {
+          folderId: '',
+          lockId: '',
+        },
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.refreshFolderLock(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2328]: and here

File: tests/controllers/storage.test.ts [hunk 12]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        params: {
+          folderId: '',
+          lockId: '',
+        },
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.releaseFolderLock(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2427]: and here

File: tests/controllers/storage.test.ts [hunk 13]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        },
+        headers: {
+          'internxt-mnemonic': '4'
+        }
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.renameFileInNetwork(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2528]: and here

File: tests/controllers/storage.test.ts [hunk 14]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2553]: here too

File: tests/controllers/storage.test.ts [hunk 15]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2578]: and here

File: tests/controllers/storage.test.ts [hunk 16]
```diff
@@ -1400,6 +1400,1371 @@ describe('Storage controller', () => {
...
+        },
+        headers: {
+          'internxt-mnemonic': ''
+        }
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.renameFileInNetwork(request, response);
```
Comment (sg-gs) [inline tests/controllers/storage.test.ts:2603]: and here

File: tests/controllers/share.test.ts [hunk 17]
```diff
@@ -822,6 +822,77 @@ describe('Share controller', () => {
...
+        params: {
+          shareId: '',
+          folderId: ''
+        },
+      }) as unknown as Request<{ shareId: string, folderId: string }>;
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.getSharedFolderSize(request, response);
```
Comment (sg-gs) [inline tests/controllers/share.test.ts:840]: `expect(true).to.be.false`

**Review by sg-gs — CHANGES_REQUESTED** [review]

**Comment by sg-gs** [comment]
Great to see some testing coverage on our core product backend

File: src/app/routes/storage.ts (hunk 1 above)
Comment (sraver) [reply src/app/routes/storage.ts:368]: idk.. i wanted to modify the logic inside the handler as little as possible, so I just left it was

Comment (sraver) [reply src/app/routes/storage.ts:368]: should i change it? So far it works..

**Review by sg-gs — APPROVED** [review]

File: src/app/routes/storage.ts (hunk 1 above)
Comment (sg-gs) [reply src/app/routes/storage.ts:368]: I left this comment just as a thought of this legacy code that we would change once Desktop old users upgrade from 1.5. The goal - which has been accomplished very good - was to left anything as it is but in TypeScript, do not worry

**Pull request merged**